jujugui/static/gui/src/app/init.js

```diff
@@ -0,0 +1,1590 @@
+// /* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+/*
+  In an effort to develop this init.js along side with the rest of the
+  app here are the changes you'll have to manually change to continue
+  working on the new system.
+
+  index.html.mako
+    - Uncomment the init-pkg.js script tags lines 309, 318.
+    - Uncomment the initialization code lines 358, 359.
+    - Comment the old init code lines 357, 362.
+  Makefile
+    - Uncomment the hack to generate the init-pkg file Line 212.
+
+  Code to still move over from app.js
+    - Y.juju.Cookies Line 50.
+    - widgets.AutodeployExtension Line 49.
+*/
+
+const mixwith = require('mixwith');
+
+const utils = require('./init/utils');
+const hotkeys = require('./init/hotkeys');
+const csUser = require('./init/charmstore-user');
+
+const ComponentRenderersMixin = require('./init/component-renderers-mixin');
+const DeployerMixin = require('./init/deployer-mixin');
+
+const yui = window.yui;
+
+class GUIApp {
+  constructor(config) {
+    /**
+      The supplied application configuration object on instantiation. Stored
+      here to avoid having to pass it around.
+      @type {Object}
+    */
+    this.applicationConfig = config;
+    /**
+      Holds the modelUUID the application should connect to. It's possible that
+      this value and the value in the modelAPI instance will differ as another
+      portion of the application has set this value prior to switching models.
+      @type {String}
+    */
+    this.modelUUID = config.jujuEnvUUID;
+    /**
+      The default web page title.
+      @type {String}
+    */
+    this.defaultPageTitle = 'Juju GUI';
+    /**
+      The object used for storing a mapping of previously visited user paths
+      to the type of entity (model, store). e.g. /u/spinach/ghost would map to
+      store.
+      @type {Map}
+    */
+    this.userPaths = new Map();
+    /**
+      Track anonymous mode. This value will be set to true when anonymous
+      navigation is allowed, in essence when a GISF anonymous user is being
+      modeling on a new canvas.
+      @type {Boolean}
+    */
+    this.anonymousMode = false;
+    /**
+      Reference to the changesUtils utilities.
+      @type {Object}
+    */
+    this.changesUtils = window.juju.utils.ChangesUtils;
```

Comment:
Definitely can and should, however, because it's still a YUI module and I can't require it, I didn't want to include it in multiple places. I'll make a note to do just that when it's de-yui'd